bentoml/_internal/configuration/__init__.py

```diff
@@ -52,39 +41,20 @@ def clean_bentoml_version(bentoml_version: str) -> str:
 
 
 # Find BentoML version managed by setuptools_scm
-BENTOML_VERSION: str = importlib_metadata.version("bentoml")
+BENTOML_VERSION: str = get_pkg_version("bentoml")
 # Get clean BentoML version indicating latest PyPI release. E.g. 1.0.0.post => 1.0.0
 CLEAN_BENTOML_VERSION: str = clean_bentoml_version(BENTOML_VERSION)
 
 
 @lru_cache(maxsize=1)
 def is_pypi_installed_bentoml() -> bool:
-    """Returns true if BentoML is installed via PyPI official release or installed from
-     source with a release tag, which should come with pre-built docker base image on
-     dockerhub.
-
-    BentoML uses setuptools_scm to manage its versions, it looks at three things:
-
-    * the latest tag (with a version number)
-    * the distance to this tag (e.g. number of revisions since latest tag)
-    * workdir state (e.g. uncommitted changes since latest tag)
-
-    BentoML uses setuptools_scm with `version_scheme = "post-release"` option, which
-    uses roughly the following logic to render the version:
-
-    * no distance and clean: {tag}
-    * distance and clean: {tag}.post{distance}+{scm letter}{revision hash}
-    * no distance and not clean: {tag}+dYYYYMMDD
-    * distance and not clean: {tag}.post{distance}+{scm letter}{revision hash}.dYYYYMMDD
-
-    This function looks at the version str and decide if BentoML installation is
-    base on a recent official release.
-    """
-    # In a git repo with no tag, setuptools_scm generated version starts with "0.1."
-    is_tagged = not BENTOML_VERSION.startswith("0.1.")
-    is_clean = not str(version_mod.version_tuple[-1]).split(".")[-1].startswith("d")
-    not_been_modified = BENTOML_VERSION == BENTOML_VERSION.split("+")[0]
-    return is_tagged and is_clean and not_been_modified
+    # We can use importlib.util.find_spec("bentoml").submodule_search_locations
+    # to find the source code of bentoml. If bentoml is installed with pypi, then it should
+    # be installed in 'site-packages', otherwise, it will return path to source git.
+    bentoml_path = source_locations("bentoml")
+    if bentoml_path and "site-packages" in bentoml_path:
```

Comment:
this doesn't mean BentoML is installed from pypi index, it can be installed from source - this logic here might be used for checking if it's in edit mode tho

Reply:
if it is installed in edit mode, then `submodule_search_locations` returns the path to the directory.

Reply:
This will come with a bug where if users install from source with the latest release tags, it would match with the same tag of the one released on pypi

Reply:
> This will come with a bug where if users install from source with the latest release tags, it would match with the same tag of the one released on pypi

That is expected behavior - our release process ensures the tags are matching PyPI releases. why is that a problem?